src/main/java/ch/njol/skript/conditions/CondIsTimezoneValid.java

```diff
@@ -0,0 +1,62 @@
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.time.DateTimeException;
+import java.time.ZoneId;
+
+@Name("Is Timezone Valid")
+@Description("Checks if a timezone is valid.")
+@Example("""
+	set {_timezone} to "America/New_York"
+	if timezone {_timezone} is valid:
+		set {_date} to now in timezone {_timezone}
+	""")
+@Since("INSERT VERSION")
+public class CondIsTimezoneValid extends Condition {
+
+	static {
+		Skript.registerCondition(CondIsTimezoneValid.class, "time[ ]zone[s] %strings% (is|are) [negate:in]valid");
+	}
+
+	private Expression<String> timezones;
+	private boolean isNegated;
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		timezones = (Expression<String>) expressions[0];
+		isNegated = parseResult.hasTag("negate");
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		for (String timezone : timezones.getAll(event)) {
```

Comment:
```suggestion
		for (String timezone : timezones.getArray(event)) {
```
Don't use getAll unless you have a specific reason to. It treats OR lists like AND lists